### DISCUSSION TITLE
Perhaps a function that recursively returns itself

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1175] [General]
let rec Job configuration (requiredObject : SomeType) = 
    //do something
    (fun (e : SomeType) -> Job configuration e)

I was redirected by the folks over C# discussions to here, I looked at a few discussions and I felt appropriate to post language suggestions and sometimes maybe questions on this repository that has the word suggestions in the name. I appreciate the pointer to Stack Overflow for answers, but unfortunately I don't speak english while china is blocking a lot of the websites, so I am going to post my discussions here, unless of course, I am not welcomed in this community by simply suggesting the language features

Look, folks, here's the thing : Job configuration <| ``Do Something with This`` <| ``Do Something with That``

**Comment by Xyncgas** [comment]
Another way of looking at these paste few day of, what I have been playing with
Currying can definitely turn functions into one parameter function
but to then apply the function on its parameter we have to call the function again and again
what if you are building a logger, do you like logging a bunch of message by calling a series of logger.log, or do you maybe want to pipe a bunch (unlimited) of string to the logger
It becomes obvious at this point that it can be an improvement on the technology

**Comment by Xyncgas** [comment]
Since f# has first class function, it's important that we are able to do this

**Comment by Xyncgas** [comment]
Because we can defined recursive types and records and use it as the return type to unwrap itself, however the same thing we were to do with functions themselves, let me just say there doesn't seem to be a way

**Comment by Xyncgas** [comment]
gives me a feeling that f# isn't a matured language

**Comment by vzarytovskii** [comment]
This repository is **only** for language suggestions, not discussions and questions, if you don't have access to StackOverflow, please post your questions in https://github.com/dotnet/fsharp/discussions, people there will help you to formulate a suggestion if necessary, thanks.

**Reply by SchlenkR** [reply]
> This repository is only for language suggestions, not discussions and questions

I guess that applies to the issues section of this repo, not the discussion section. In #1154 @dsyme wrote about the discussion section:

> We’re using Discussions as a place to connect with other members of our community. We hope that you: / Ask questions you’re wondering about. / Share ideas. / Engage with other community members. / Welcome others and are open-minded. Remember that this is a community we build together

**Reply by vzarytovskii** [reply]
> > This repository is only for language suggestions, not discussions and questions
> 
> I guess that applies to the issues section of this repo, not the discussion section. In #1154 @dsyme wrote about the discussion section:
> 
> > We’re using Discussions as a place to connect with other members of our community. We hope that you: / Ask questions you’re wondering about. / Share ideas. / Engage with other community members. / Welcome others and are open-minded. Remember that this is a community we build together 

I guess we then should leave one only of them, otherwise information will be too fragmented?

**Reply by SchlenkR** [reply]
Do you mean we should close the discussion section - or could you clarify what you mean?

**Reply by vzarytovskii** [reply]
> Do you mean we should close the discussion section - or could you clarify what you mean?

I'd say we should have one place we point people for discussions, instead of two - either here or dotnet/fsharp. I think it will be slightly less confusing.

**Reply by Xyncgas** [reply]
There is no discussion in fsharp/fsharp repository, since there is two repository one for fsharp language design the other's fsharp suggestion I would say this is the place

Also since fsharp if its own thing (even though its got strings with microsoft) would it be nicer to keep things here rather than in microsoft's repository, see fsharp even got its own ionide IDE

**Reply by vzarytovskii** [reply]
> There is no discussion in fsharp/fsharp repository, since there is two repository one for fsharp language design the other's fsharp suggestion I would say this is the place
> 

Up to you then, but trust me, it's even less here.
People rarely look at discussions in general unfortunately.

**Comment by Xyncgas** [comment]
After some researching & playing & trying I found a way to make it work (ran in f#interactive)
```
    let rec k f (a:string) :obj=
        f(a)
        k f
    let inline (|>) x (y:obj) =
        let j =  y :?> ('a->obj)
        x |> j
    let inline (<|) (x:obj) y  =
        let j = x :?> ('a->obj)
        j <| y

    let z =  k (System.Console.WriteLine) "Hello" <| "World" <| "People"
```
The problem is I can't have however many parameters for k it's hard coded, also I had to change the operators myself, it would be nice if the language can use code generator to generate these ``` glue codes ``` it would be nice if the compiler can check that for example I am using the original operator in the overloaded operator's body
